adagrams/game.py

```diff
@@ -1,11 +1,142 @@
+    'S': 1,
+    'T': 1,
+    'U': 1,
+    'V': 4,
+    'W': 4,
+    'X': 8,
+    'Y': 4,
+    'Z': 10
+}
+
+
 def draw_letters():
-    pass
+    ## pool_of_letters_array is an array that will store the letters and the number of times it occurs
+    pool_of_letters_array = []
+
+    ## for loop: iteration through LETTER_POOL to combine subset list into the pool_of_letters_array list
+    for letter in LETTER_POOL: 
+        pool_of_letters_array += list(letter * LETTER_POOL[letter])
+    
+    array_of_letters = pool_of_letters_array.copy()
+
+    ## the use of while and if conditions ensures 10 random uniquely selected letters
+    letter_list = []
+
+    while len(letter_list) < 10:
+        random_letter = random.choice(array_of_letters)
+        letter_list.append(random_letter)
+        array_of_letters.remove(random_letter)
+        
+    array_of_letters = pool_of_letters_array.copy()
+
+    return letter_list
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word_alpha_list = []
+    letter_bank_copy = letter_bank.copy()
+
+    ## for loop: iteration through letter_bank_copy and popping out letter that appears in our hand to keep track of the letter frequency and how many times we can draw it
+    for letter in word:
+        if letter in letter_bank_copy:
+            popped_letter = letter_bank_copy.pop(letter_bank_copy.index(letter))
+            word_alpha_list.append(popped_letter)
+            
+    ## checks for if the word guessed is using letters from inside our drawn hand (word_alpha_list)
+    if len(word_alpha_list) == len(word):
+            return True
+
+    elif letter not in letter_bank_copy and not letter.isupper():
+        if len(word_alpha_list) < len(word):
+            return word, letter_bank_copy
+
+    else:
+        return False
+
+    return word, letter_bank
+
 
 def score_word(word):
-    pass
+    BONUS_POINTS = 8
+    points = 0
+
+    ## for loop: iteration through each letter in the guessed word for a score. the guessed word is forced to be uppercase to be able to score the letter.
+
+    for letter in word.upper():
+        points += SCORE_CHART[letter]
+
+    if len(word) >= 7 and len(word) <= 10:
+        points += BONUS_POINTS
+
+    return points
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    highest_word = word_list[0]
+    highest_score = score_word(word_list[0])
+
+    ## scores word from first index to the end of the word_list, to compare against the word and score from index 0
+    for word in word_list[1::]:
+        current_score = score_word(word)
+    
+        if current_score > highest_score:
+            highest_score = current_score
+            highest_word = word
+        elif current_score == highest_score:
+            if len(highest_word) == 10:
+                continue
+            elif len(word) == 10:
+                highest_word = word
+            elif len(word) < len(highest_word):
+                highest_word = word
```

Comment:
Both of the `elif` statements have the same contents `highest_word = word`. Repeating code like this is often a sign that we should look at our code to see if there are ways we could combine our statements to reduce repetition. Thinking about long-term code maintenance, if we have to change that code in the future, it goes quicker and is less error-prone when there is only one line to update, rather than several lines that we need to keep in sync.  

What could it it look like to combine these statements?